sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,412 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import logging
+
+from daal4py.sklearn._utils import daal_check_version, sklearn_check_version
+
+if daal_check_version((2024, "P", 600)):
+    import numpy as np
+    from scipy.sparse import issparse
+    from sklearn.linear_model import Ridge as sklearn_Ridge
+    from sklearn.metrics import r2_score
+    from sklearn.utils.validation import check_is_fitted, check_X_y
+
+    from daal4py.sklearn.linear_model._ridge import _fit_ridge as daal4py_fit_ridge
+
+    if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+        from sklearn.linear_model._base import _deprecate_normalize
+
+    from onedal.linear_model import Ridge as onedal_Ridge
+    from onedal.utils import _num_features, _num_samples
+
+    from ..._device_offload import dispatch, wrap_output_data
+    from ..._utils import PatchingConditionsChain
+    from ...utils import get_namespace
+    from ...utils.validation import _assert_all_finite
+
+    def _is_numeric_scalar(value):
+        """
+        Determines if the provided value is an instance of either int or float.
+
+        Args:
+        value: The value to be checked.
+
+        Returns:
+        bool: True if the value is either an int or a float, False otherwise.
+        """
+        return isinstance(value, (int, float))
+
+    def _test_type_and_finiteness(X_in):
+        """
+        Checks if the input is of a supported type and is finite.
+
+        Args:
+        X_in: The input to be checked.
+
+        Returns:
+        bool: True if the input is of a supported type and is finite, False otherwise.
+        """
+
+        xp, _ = get_namespace(X_in)
+        X = xp.asarray(X_in)
+
+        if np.iscomplexobj(X):
+            return False
+
+        try:
+            _assert_all_finite(X)
+        except BaseException:
+            return False
+
+        return True
```

Comment:
shouldn't this be Ridge class method? Is it required to be free func?

Reply:
does not use anything related to class object, why make it a method? i would consider moving it to general utils that can be reused across different modules

Reply:
My comment based on your previous replies. https://github.com/intel/scikit-learn-intelex/pull/1843#discussion_r1656944833. And here it is a method. 
Previously my comment on moving this into utils module was ignored, so I decided to clarify.

If this is going to be common utility than I am totally ok with this.

Reply:
I see that discussion the productive discussion about this code continued at https://github.com/intel/scikit-learn-intelex/pull/1843#discussion_r1670156726. Closing this one as resolved while.